deployer/pom.xml

```diff
@@ -0,0 +1,131 @@
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.maven.shared</groupId>
+            <artifactId>maven-invoker</artifactId>
+            <version>2.2</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-compiler-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.rat</groupId>
+                <artifactId>apache-rat-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <argLine>-Dmsf4jTestRunning</argLine>
+                    <suiteXmlFiles>
+                        <suiteXmlFile>src/test/resources/testng.xml</suiteXmlFile>
+                    </suiteXmlFiles>
+                    <systemPropertyVariables>
+                        <maven.home>${maven.home}</maven.home>
+                        <project.filepath>${project.basedir}</project.filepath>
+                    </systemPropertyVariables>
+                </configuration>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-deploy-plugin</artifactId>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.felix</groupId>
+                <artifactId>maven-bundle-plugin</artifactId>
+            </plugin>
+        </plugins>
+    </build>
+
+    <properties>
+        <private.package>org.wso2.msf4j.deployer.internal</private.package>
+        <export.package>
+            !org.wso2.msf4j.deployer.internal,
+            org.wso2.msf4j.deployer;version="${msf4j.version}"
+        </export.package>
+        <import.package>
+            org.wso2.msf4j.*;version="${msf4j.version}",
+            javax.annotation.*,
+            javax.ws.rs.*,
+            org.slf4j.*;version="${slf4j.version.range}",
+            org.osgi.framework.*;version="${osgi.framework.import.version.range}",
+            org.wso2.carbon.kernel.startupresolver.*;version="${carbon.kernel.version.range}",
+            org.wso2.carbon.deployment.engine.*;version="${carbon.deployment.version.range}",
+        </import.package>
+        <carbon.component>
+            startup.listener;componentName="wso2-microservices-deployer";requiredService="org.wso2.msf4j.MicroservicesRegistry",
+            osgi.service; objectClass="org.wso2.carbon.deployment.engine.Deployer"
+        </carbon.component>
+        <dynamic.import.package>*</dynamic.import.package>
```

Comment:
Do we need `<dynamic.import.package>*</dynamic.import.package>`? IMO we should remove this line

Reply:
We need this since this will require to load, spring annotations and dependencies.

Reply:
In that case, we should add a comment explaining so IMO